File: zmusic-bungee/src/main/java/me/zhenxin/zmusic/ZMusicBungee.java
package me.zhenxin.zmusic;

import me.zhenxin.zmusic.dependencies.annotation.RuntimeDependency;
import me.zhenxin.zmusic.platform.Platform;
import me.zhenxin.zmusic.platform.impl.LoggerBungee;
import net.md_5.bungee.api.plugin.Plugin;
import org.bstats.bungeecord.Metrics;

/**
 * BungeeCord 入口类
 *
 * @author 真心
 * @email qgzhenxin@qq.com
 * @since 2023/8/28 12:26
 */
@SuppressWarnings("AlibabaClassNamingShouldBeCamel")
@RuntimeDependency(
        value = "!org.bstats:bstats-bungeecord:" + ZMusicConstants.BSTATS_VERSION,
        test = "!me.zhenxin.zmusic.library.bstats.bungeecord.Metrics",
        relocate = {"!org.bstats.", "!me.zhenxin.zmusic.library.bstats."}
)
public class ZMusicBungee extends Plugin {

    @Override
    public void onLoad() {
        ZMusicRuntime.setup(getDataFolder().getAbsolutePath(), getLogger(), ZMusicBungee.class);
    }

    @Override
    public void onEnable() {
        ZMusicKt.setLogger(new LoggerBungee(getProxy().getConsole()));
        ZMusicKt.setDataFolder(getDataFolder());
        ZMusicKt.setPlatform(Platform.BUNGEE);
        new Metrics(this, 8864);
        ZMusic.onEnable();
    }

    @Override
    public void onDisable() {
        ZMusic.onDisable();
    }
}

File: bungee/src/main/kotlin/me/zhenxin/zmusic/bungee/ZMusicBungee.kt
package me.zhenxin.zmusic.bungee

import me.zhenxin.zmusic.common.ZMusic
import me.zhenxin.zmusic.common.modules.logger.Logger
import net.md_5.bungee.api.plugin.Plugin

/**
 * ZMusic Bungeecord 主入口
 *
 * @author 真心
 * @since 2021/7/8 13:56
 * @email qgzhenxin@qq.com
 */
class ZMusicBungee : Plugin() {
    override fun onEnable() {
        me.zhenxin.zmusic.common.logger = Logger(logger)
        MetricsBC(this, 8864)
        proxy.pluginManager.registerCommand(this, CmdExec())
        ZMusic.onEnable(dataFolder)
    }
}
